src/components/VideoCards/VideoCards.jsx

```diff
@@ -24,47 +26,40 @@ export default function VideoCard({ YouTubeLink, videoTitle, videoId, bookmark,
 		setIsBookmarked(id);
 	}
 
-	function getYouTubeId(url) {
-		url = url.split(/(vi\/|v=|\/v\/|youtu\.be\/|\/embed\/|\/shorts\/)/);
-		return url[2] !== undefined ? url[2].split(/[^0-9a-z_-]/i)[0] : url[0];
-	}
-
-	const videoID = getYouTubeId(YouTubeLink);
+	const videoID = useYouTubeId(YouTubeLink);
 
 	return (
 		<StyledVideoContainer key={videoId} data-testid="videoContainer">
 			<StyledVideoButtonContainer>
 				<StyledButton type="button" variant="videoIcons" onClick={handleVisibility}>
 					<Icon variant={isShown ? 'deleteFilled' : 'delete'} color="white" />
 				</StyledButton>
+
 				{isShown ? <DeleteModal onCancel={handleVisibility} videoId={videoId} /> : ''}
+
 				<Link href={`/edit/${videoId}`}>
-					<a>
-						<StyledButton type="button" variant="videoIcons">
-							<Icon variant="edit" color="white" />
-						</StyledButton>
-					</a>
+					<StyledLink>
+						<Icon variant="edit" color="white" />
+					</StyledLink>
```

Comment:
I'd add an `aria-label` here: 
```suggestion
					<StyledLink aria-label="edit">
						<Icon variant="edit" color="white" />
					</StyledLink>
```